packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/link-table/use-row-data.ts

```diff
@@ -0,0 +1,41 @@
+import { useMemo } from "react";
+import { LinkRow } from "./types";
+import { useEntityEditor } from "../entity-editor-context";
+import { generateEntityLabel } from "../../../../../lib/entities";
+import { sortRowData } from "../../../../../components/GlideGlid/utils";
+
+export const useRowData = () => {
+  const { entity, linkSort } = useEntityEditor();
+
+  const rowData = useMemo<LinkRow[]>(() => {
+    if (!entity) {
+      return [];
+    }
+
+    return (
+      entity?.links.map((link) => {
+        const linkType =
+          entity.entityTypeRootedSubgraph.referencedLinkTypes.find(
+            (val) => val.linkTypeId === link.linkTypeId,
+          )?.linkType;
+
+        const referencedEntityType =
+          entity.entityTypeRootedSubgraph.referencedEntityTypes.find(
+            (val) => val.entityTypeId === link.targetEntity.entityTypeId,
+          )?.entityType.title;
+
+        return {
+          expectedEntityType: referencedEntityType ?? "",
+          linkedWith: generateEntityLabel(link.targetEntity),
+          linkId: link.linkTypeId,
+          relationShip: "Outbound",
+          type: linkType?.title ?? "",
+        };
+      }) ?? []
+    );
+  }, [entity]);
+
+  const sortedRowData = sortRowData(rowData, linkSort);
```

Comment:
This looks like it ought to be memo'd. It is for property tables, and this ends up in the dependency array of `useCallback` in `useGetCellContent`, so really ought to be stable

Reply:
Ugh, thanks for catching this, there are two `sortedRowData`'s for link table & entity table. I forgot to memoize this one 🤦‍♂️